askup/views.py

```diff
@@ -236,63 +228,30 @@ def get_queryset(self):
 
 
 @login_required
-def user_profile_view(request, user_id):
+def user_profile_view(request, user_id, organization_id=None):
     """Provide the user profile my questions view."""
     profile_user = get_object_or_404(User, pk=user_id)
     user_id = int(user_id)
+    selected_organization = select_user_organization(profile_user, organization_id)
     return render(
         request,
         'askup/user_profile.html',
-        {
-            'profile_user': profile_user,
-            'full_name': compose_user_full_name_from_object(profile_user),
-            'viewer_user_id': request.user.id,
-            'own_score': get_user_score_by_id(user_id),
-            'is_owner': user_id == request.user.id,
-            'is_student': check_user_has_groups(profile_user, 'student'),
-            'own_correct_answers': get_user_correct_answers_count(user_id),
-            'own_incorrect_answers': get_user_incorrect_answers_count(user_id),
-            'user_rank_place': get_user_place_in_rank_list(user_id),
-            'own_last_week_questions': get_student_last_week_questions_count(user_id),
-            'own_last_week_thumbs_up': get_student_last_week_votes_value(user_id),
-            'own_last_week_correct_answers': get_student_last_week_correct_answers_count(user_id),
-            'own_last_week_incorrect_answers': get_student_last_week_incorrect_answers_count(user_id),
-            'user_organizations': get_user_organizations_string(profile_user),
-            'rank_list': tuple(),
-            'rank_list_total_users': 0,
-            'own_subjects': get_user_subjects(user_id),
-        },
+        get_user_profile_context_data(request, profile_user, user_id, selected_organization),
     )
 
 
 @login_required
-def user_profile_rank_list_view(request, user_id):
+def user_profile_rank_list_view(request, user_id, organization_id=None):
     """Provide the user profile rank list view."""
```

Comment:
I would rewrite to the multiple docstrings

Reply:
done